src/Parallel/Printf.hpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines Parallel::printf for writing to stdout
+
+#pragma once
+
+#include <charm++.h>
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+namespace Parallel {
+namespace detail {
+// Print pointers
+template <typename T,
+          typename std::enable_if_t<
+              std::is_fundamental<std::decay_t<
+                  std::remove_pointer_t<std::decay_t<T>>>>::value or
+              std::is_pointer<T>::value> * = nullptr>
+inline constexpr T convert_to_printable_type(T &&t) {
+  return t;
+}
+
+// Print objects with stream operator
+template <typename T,
+          typename std::enable_if_t<
+              not std::is_fundamental<std::decay_t<
+                  std::remove_pointer_t<std::decay_t<T>>>>::value and
+              not std::is_pointer<T>::value> * = nullptr>
+inline std::string convert_to_printable_type(T &&t) {
+  std::stringstream ss;
```

Comment:
I'll try it, don't see why we couldn't other than that I didn't want to include type traits until the next PR, but I have the code in on a different branch so I could pull that in no problem if you're okay with increasing the PR size